enterprise/reporting/error_index_reporting.go

```diff
@@ -0,0 +1,157 @@
+package reporting
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+	"github.com/rudderlabs/rudder-server/utils/types"
+)
+
+const (
+	errorIndex = "err_idx"
+)
+
+type payload struct {
+	MessageID        string    `json:"messageId"`
+	SourceID         string    `json:"sourceId"`
+	DestinationID    string    `json:"destinationId"`
+	TransformationID string    `json:"transformationId"`
+	TrackingPlanID   string    `json:"trackingPlanId"`
+	FailedStage      string    `json:"failedStage"`
+	EventType        string    `json:"eventType"`
+	EventName        string    `json:"eventName"`
+	ReceivedAt       time.Time `json:"receivedAt"`
+	FailedAt         time.Time `json:"failedAt"`
+}
```

Comment:
@cisse21 do these cover our own current needs?

Reply:
Yeah this looks good for all the current needs